### PR TITLE
Refactored getCurrentUser.js and setCookie

### PR DESCRIPTION
**Pull request by nikos467**
Made the export function easier  to read

### CODE REVIEW

File: lib/user/setCookie.js [hunk 1]
```diff
@@ -18,22 +18,20 @@ exports.optional = ['validate']
...
-      .then(function (res) {
-        options.jar.session = args.cookie
-        return res
-      }).catch(function (e) {
-        console.error('Failed to validate cookie: Are you sure the cookie is valid?\nEnsure you include the full cookie, including warning text.')
-        throw new Error(e)
-      })
+  }
+  try {
+    const res = await getCurrentUser({ jar: { session: args.cookie } })
```
Comment (Neztore) [inline lib/user/setCookie.js:31]: Could `return await getCurrentUser(...);` to remove an unnecessary line and variable.

Comment (Neztore) [reply lib/user/setCookie.js:31]: Or, a better solution would be to match the original behaviour and actually set the session token following validation!

Please test your PRs. Reading your code, if validate is enabled the internally stored cookie will never be updated.

**Review by Neztore — CHANGES_REQUESTED** [review]
Please edit your PR to set the internally stored token in jar

**Comment by Chosey98** [comment]
Matter of fact I just reviewed and pushed the changes @Neztore requested to @nikos467 branch and waiting his approval to merge

**Comment by Neztore** [comment]
No activity and fixed elsewhere

**Pull request closed without merging**